estate/models/estate_property.py

```diff
@@ -0,0 +1,99 @@
+
+from dateutil.relativedelta import relativedelta
+from odoo import fields, models, api
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools.float_utils import float_compare
+
+
+class Properties(models.Model):
+    _name = "estate.property"
+    _description = "Properties of the estate model"
+    _order = "id desc"
+
+    active = fields.Boolean(default=True)
+    name = fields.Char("Title", required=True)
+    description = fields.Text("description")
+    postcode = fields.Integer("Post code")
+    date_availability = fields.Date(
+        "Date availability",
+        copy=False,
+        default=fields.Date.today() + relativedelta(months=3))
+    expected_price = fields.Float("Expected price", required=True)
+    selling_price = fields.Float("Selling price", readonly=True, copy=False)
+    bedrooms = fields.Integer("# Bedrooms", default=2)
+    living_area = fields.Integer("Living area")
+    facades = fields.Integer("# facades")
+    garage = fields.Boolean("Garage")
+    garden = fields.Boolean("Garden")
+    garden_area = fields.Integer("Garden area", default=0)
+    garden_orientation = fields.Selection(
+        string="Orientation",
+        selection=[("north", "North"), ("east", "East"), ("west", "West"), ("south", "South")])
+    state = fields.Selection(
+        selection=[
+            ("new", "New"),
+            ("offer received", "Offer Received"),
+            ("offer accepted", "Offer Accepted"),
+            ("sold", "Sold"),
+            ("cancelled", "Cancelled")],
+        required=True,
+        copy=False,
+        default="new")
+    property_type_id = fields.Many2one("estate.property.type")
+    buyer = fields.Many2one("res.partner", copy=False)
+    salesperson = fields.Many2one("res.users", default=lambda self: self.env.user)
+    tag_ids = fields.Many2many("estate.property.tag")
+    offer_ids = fields.One2many("estate.property.offer", inverse_name="property_id")
+    total_area = fields.Float("Total Area", compute="_compute_area")
+    best_price = fields.Float("Best Price", compute="_find_best_price")
+
+    _sql_constraints = [
+        ('positive_expecting_price', "CHECK(expected_price > 0)", "Property should have a strictly positive expected price !"),
+        ("positive_sell_price", "CHECK(selling_price > 0)", "Property should have a strictly positive selling price !"),
+        ]
+
+    @api.depends("garden_area", "living_area")
+    def _compute_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends("offer_ids")
+    def _find_best_price(self):
+        for record in self:
+            record.best_price = max(record.offer_ids.mapped("price") + [0])
```

Comment:
```suggestion
            record.best_price = max(record.offer_ids.mapped("price"), default=0)
```
`max()` lets you set a default value